src/VisualStudio/CSharp/Test/ProjectSystemShim/CPS/SourceFileHandlingTests.cs

```diff
@@ -57,5 +58,84 @@ public void AddRemoveAdditionalFile_CPS()
                 Assert.Empty(GetCurrentAdditionalDocuments());
             }
         }
+
+        [WpfFact]
+        [Trait(Traits.Feature, Traits.Features.ProjectSystemShims)]
+        public void ReorderSourceFiles_CPS()
```

Comment:
Consider adding tests for the batching scenarios too.

Reply:
I did, look below this test.